wgpu-types/src/lib.rs

```diff
@@ -308,18 +366,33 @@ bitflags::bitflags! {
     }
 }
 
+/// Describes the properties of the device when requesting a device.
 #[repr(C)]
 #[derive(Clone, Debug, Default)]
 #[cfg_attr(feature = "trace", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 pub struct DeviceDescriptor {
+    /// Extensions that the device should support. If any extension is not supported by
+    /// the adapter, creating a device will panic.
     pub extensions: Extensions,
+    /// Limits that the device should support. If any limit is "higher" than the limit exposed by
+    /// the adapter, creating a device will panic.
     pub limits: Limits,
     /// Switch shader validation on/off. This is a temporary field
     /// that will be removed once our validation logic is complete.
     pub shader_validation: bool,
 }
 
+/// Reads something that impls [`std::io::Read`] and reads the following bytestream as SPIRV, turning
+/// it into a Vec<u32> to be handed to shader creation.
```

Comment:
"Reads SPIR-V from a readable object" - you don't need anything complicated here